etest/__init__.py

```diff
@@ -47,9 +47,10 @@ def echo_check_verbose(check):
 @click.option("-j", "--jobs", default=1, help="number of test to run simultaneously")
 @click.option("-q", "--quiet", is_flag=True, default=False, help="suppress all output")
 @click.option("-v", "--verbose", is_flag=True, default=False, help="provide more output")
+@click.option("-a", "--arch", default="amd64", help="architecture to test against")
```

Comment:
Should this be a choice instead of an arbitrary string?  It feels like making it a choice would be the safer option.

Reply:
I feel like there are too many options, and I dont know how to make them exclusive. It already does a bunch of sanitizing to make sure that it runs correctly if the user meant a correct architecture. However, even if they dont, itll just error out.

Reply:
Is that error clearly letting the user know that it's due to the architecture not existing?  If so, then I say it's good enough.  I also assumed we only had a limited set of supported architectures.  If that's the case the choice makes way more sense.  AS for making them exclusive that's easy with click's choice type: https://click.palletsprojects.com/en/7.x/options/#choice-options

Reply:
I thought that by "choice" you meant something along the lines of ``--amd64 --arm64 --ppc64 --x86`` etc.
The model you sent does seem a lot more sensible and makes a lot of sense for the usecase.